Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.AzureAD.UI;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Identity.Web.TokenCacheProviders.Distributed
+{
+    /// <summary>
+    /// An implementation of token cache for both Confidential and Public clients backed by MemoryCache.
+    /// </summary>
+    /// <seealso cref="https://aka.ms/msal-net-token-cache-serialization"/>
+    public class MsalDistributedTokenCacheProvider : MsalAbstractTokenCacheProvider
+    {
+        /// <summary>
+        /// .NET Core Memory cache
+        /// </summary>
+        private readonly IDistributedCache _distributedCache;
+
+        /// <summary>
+        /// Msal memory token cache options
+        /// </summary>
+        private readonly DistributedCacheEntryOptions _cacheOptions;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="azureAdOptions"></param>
+        /// <param name="httpContextAccessor"></param>
+        /// <param name="memoryCache"></param>
+        /// <param name="cacheOptions"></param>
+        public MsalDistributedTokenCacheProvider(IOptions<AzureADOptions> azureAdOptions,
+                                            IHttpContextAccessor httpContextAccessor,
+                                            IDistributedCache memoryCache,
+                                            IOptions<DistributedCacheEntryOptions> cacheOptions) :
+            base(azureAdOptions, httpContextAccessor)
+        {
+            _distributedCache = memoryCache;
```

Comment:
null checks

Reply:
I don't think that this is needed @bgavrilMS , as this constructeurs are called by DI, which will have thrown if the corresponding services cannot be found.
@MarkZuber : do you confirm?